app/src/app/app-shell/app-shell.module.ts

```diff
@@ -0,0 +1,40 @@
+import { CommonModule } from '@angular/common';
+import { ModuleWithProviders, NgModule } from '@angular/core';
+import { TranslateModule } from '@ngx-translate/core';
+import { Routes, provideRoutes, RouterModule } from '@angular/router';
+import { ShellDummyGuard, SHELL_LAYOUT_ROUTE } from './app-shell.routes';
+import { BlankPageComponent, SidenavLayoutModule } from '@alfresco/adf-core';
+import { ExtensionService, ExtensionsModule, provideExtensionConfig } from '@alfresco/adf-extensions';
+import { ShellLayoutComponent } from './components/shell/shell.component';
+import { ContentModule } from '@alfresco/adf-content-services';
+
+@NgModule({
+  imports: [SidenavLayoutModule, ContentModule, ExtensionsModule, RouterModule.forChild([]), CommonModule, TranslateModule.forChild()],
+  exports: [ShellLayoutComponent],
+  declarations: [ShellLayoutComponent],
+  providers: [provideExtensionConfig(['shell.plugin.json'])]
+})
+export class AppShellModule {
+  static withChildren(childRouters: Routes[]): ModuleWithProviders<AppShellModule> {
+    const shellLayoutRoute = SHELL_LAYOUT_ROUTE;
+
+    childRouters.forEach((childRoute) => {
+      shellLayoutRoute.children.push(...childRoute);
+    });
+
+    return {
+      ngModule: AppShellModule,
+      providers: provideRoutes([shellLayoutRoute])
+    };
+  }
+
+  constructor(extensions: ExtensionService) {
+    extensions.setAuthGuards({
+      'app.shell.dummy.guard': ShellDummyGuard
```

Comment:
I think this one is useless and can be removed

Reply:
It is used for `blank.page` the issue is that, if we provide routes using our extension service without the guard, it is going to be provided automatically. Therefore we wont see e.g. blank page if we are not logged in (not sure right now if this is the correct way)

If we consider to remove the blank page, for sure we can remove the dummy guard

Reply:
Removed as discussed